packages/autocomplete-js/Autocomplete.js

```diff
@@ -111,6 +111,27 @@ class Autocomplete {
     this.updateStyle()
   }
 
+  destroy = () => {
+    document.body.removeEventListener('click', this.handleDocumentClick)
+    this.input.removeEventListener('input', this.core.handleInput)
+    this.input.removeEventListener('keydown', this.core.handleKeyDown)
+    this.input.removeEventListener('focus', this.core.handleFocus)
+    this.input.removeEventListener('blur', this.core.handleBlur)
+    this.resultList.removeEventListener(
+      'mousedown',
+      this.core.handleResultMouseDown
+    )
+    this.resultList.removeEventListener('click', this.core.handleResultClick)
+
+    this.root = null
+    this.input = null
+    this.resultList = null
+    this.getResultValue = null
+    this.onUpdate = null
+    this.renderResult = null
+    this.core.destroy()
```

Comment:
I _think_ since nothing would be referencing this object, it would get garbage collected just fine along with `this.core` object, but it may be better to play it safe ;) https://github.com/trevoreyre/autocomplete/pull/121/commits/1587742ff3a6aa6dcae6b5c36fe90d8826ef11d1 @trevoreyre